lib/pacemaker/pcmk_sched_notif.c

```diff
@@ -131,89 +131,95 @@ expand_node_list(GListPtr list, char **uname, char **metal)
     }
 }
 
+/*!
+ * \internal
+ * \brief Add a string to a space-separated string list
+ *
+ * \param[in,out] str_list  Pointer to existing string list
+ *                          (or pointer to NULL to start with an empty list)
+ * \param[in]     s         String to add
+ *
+ * \note String list will be reallocated and so must be dynamic memory
+ */
 static void
-expand_list(GListPtr list, char **rsc_list, char **node_list)
+add_to_string_list(char **str_list, const char *s)
```

Comment:
This would be worth turning into a pcmk__ function.  We're doing something similar in native_output_string in lib/pengine/native.c, and I'm sure there are other places too.  I just can't find them off the top of my head.  Even better would be a function that let you pass the separator character, since I'm sure we make comma separated lists this way all over the place too.

Reply:
LOL, looking into it, I see we already did :) pcmk__add_word()

I see potential uses for it in build_parameter_list(), crm_log_args(), expand_node_list(), and expand_list(). native_output_string() uses GString, so that would stay separate. We could just use GString everywhere rather than a custom function; not sure if that's better (#2126 is looking at GString for some other use cases).

Unfortunately the different cases use spaces differently. I'll have to see whether they can converge on a single approach or maybe add more args to the common function.

Reply:
create_level_registration_xml() looks to be the only additional possibility using comma instead of space